mender-convert-modify

```diff
@@ -280,7 +280,7 @@ done
 
 for overlay in "${overlays[@]}"; do
     log_info "Applying rootfs overlay: ${overlay}"
-    run_and_log_cmd "sudo rsync --archive --keep-dirlinks --verbose ${overlay}/ work/rootfs/"
+    run_and_log_cmd "sudo rsync --archive --keep-dirlinks --no-super --verbose ${overlay}/ work/rootfs/"
```

Comment:
While I completely agree that the current method where one has to be root is quite annoying, I'm wondering if this will work correctly. Won't this prevent the `work/rootfs` folder from containing any root-owned files? The resulting filesystem will then have the wrong file owners.

Reply:
You're right - the documentation sais something like that. My tests led to all files copied inheriting the rights of the containing folder.

Reply:
I just tested this, and it won't work correctly. Say you have an overlay which defines a home directory with directories owned by non-root users. Then it should appear like this:

```
drwxr-xr-x 3    0    0 4096 Mar 24 09:13 /home/
drwxr-xr-x 2 1000 1000 4096 Mar 24 09:13 /home/user
```

But with your change, it will appear like this:

```
drwxr-xr-x 3    0    0 4096 Mar 24 09:13 /home/
drwxr-xr-x 2    0    0 4096 Mar 24 09:13 /home/user
```

Reply:
I agree that that is not correct - neither is the way it works today. Do you have any ideas?

Reply:
Unfortunately I don't. I've been thinking that maybe we can do something with fakeroot, but I don't think it works together with Docker.

An alternative that the user can do is to set up the owners correctly using a hook instead of an overlay (or in addition to an overlay). That would execute inside Docker, where you are already root and can do whatever you want. But it is not automatic, of course.

Reply:
Just thinking: how about mounting overlays with overlayfs?

Reply:
Doesn't overlayfs have the same problem? I don't think it's possible to manipulate owners without being root, even on overlayfs.